### ISSUE TITLE
Polarity it is not working 

### ISSUE THREAD

**Issue by erick-repo** [cyhex/streamcrab#4]
Hi I have build successfully everything i have been able to create the databases with the following commands 
To Build Training Dataset

python collector/trainer/twitterCollector.py 
python collector/trainer/tweetClassifier.py

I have already created this files in /data dir

-rw-r--r-- 1 root root  293204 Oct 11 20:53 mood_traing_150k_1k_0.6.dat
-rw-r--r-- 1 root root 6104719 Oct 11 20:52 tweets_negative_raw.dat
-rw-r--r-- 1 root root 6581580 Oct 11 20:47 tweets_positive_raw.dat

My service it is OK

domU-12-31-39-06-8E-37 tracker # ./moodClassifierd.py start
starting...
OK

tests # more moodClientServerTest.py
# -_\- coding: utf-8 -_-

import sys
sys.path.append('../../')
from tracker.lib.moodClassifierClient import MoodClassifierTCPClient

MCC = MoodClassifierTCPClient('127.0.0.1',6666)

test_data = {'text':'I am sad because i have a bad iphone So the  4S is announced yet preorder is sold out? alright then'}

print MCC.classify(test_data, 'search')

OUTPUT it is   

---

[{'text': u'I am sad because i have a bad iphone So the  4S is announced yet preorder is sold out? alright then', 'x_mood': 0.0, 'x_lang': 'en'}]

I'm not able to identify if the polarity it is "POSITIVE or NEGATIVE" What im doing wrong because it look neutral, I am assuming that i will have to get -.xxx for negative and close to 1 for positive ?

**Comment by cyhex** [comment]
well there are couple of things you can do 

1 - play with the amount of tweets that are collected 

2 - increase cls.limit['en'] to 500k in RawClassifier or even 1M depends if you system can take it.

3 - reduce the p2_f_limit in RawClassifier, start going down in 0.05 steps - you can go down all the way to 0.5, however that also means that you DB will grow exponentially.  

4 - reduce the dumping  in MoodDetect (lib/mood_detection.py)

But the most important is to remember that this tool is based on Naive Bayes classifier ( http://en.wikipedia.org/wiki/Naive_Bayes_classifier ) by design it has a ~20% error rate. Therefore a classification of singe text will not necessary give you an expected output, however when larger sets are been classified more accurate results emerge.
In other words , if you like to test your dataset you need to test it against large amount of tweets. For my test i used per-classified 300k tweets.

**Comment by dwmcqueen** [comment]
Have you tried some of the suggestions?  I get the same issue, but perhaps it is that the initial instructions need some revisions.

**Comment by dwmcqueen** [comment]
Looking at tracker/lib/mood_detection.py we noticed these lines:
        if max(p_score,n_score) <= self.dumping:
            return 0.0

If dumping is changed, it seems to work (although results are skewed).  

Here is a test script to address this:

import sys

from tracker.lib.mood_detection import MoodDetect, MoodDetectTrainer
print 'Initializing...'
t = MoodDetectTrainer()
d = MoodDetect(t)
d.setUpSearchPhraseCleaner('')
d.dumping = 0.1
while True:
    print 'Enter some text:'
    try:
        text = raw_input()
    except EOFError:
        sys.exit(0)
    print 'mood:', d.classify(unicode(text), 'en')
